spec/app/ics-020-fungible-token-transfer/README.md

```diff
@@ -256,29 +396,63 @@ function sendFungibleTokens(
 
 ```typescript
 function onRecvPacket(packet: Packet) {
-  FungibleTokenPacketData data = packet.data
+  FungibleTokenPacketData data = unmarshal(packet.data)
   // construct default acknowledgement of success
   FungibleTokenPacketAcknowledgement ack = FungibleTokenPacketAcknowledgement{true, null}
   prefix = "{packet.sourcePort}/{packet.sourceChannel}/"
   // we are the source if the packets were prefixed by the sending chain
   source = data.denom.slice(0, len(prefix)) === prefix
+  // if we are forwarding the tokens along, we will move the coins to the forwardEscrowAddress to the source chain
+  // otherwise send directy to receiver defined by packet
+  if forwardingPath != "" {
+    // pass in denomination before receive to getNextHop
+    // i.e. denomination in packet
+    forwardPort, forwardChannel = getNextHop(forwardingPath, packet.denomination)
+    receiver = forwardEscrowAddress[forwardChannel]
```

Comment:
Couldn't we just use here `channelEscrowAddresses`? Since the `forwardChannel` is the source channel for the packet that needs to be sent to the next hop, we could use the same escrow address as for any regular packet sent from this chain on that channel, right?